### ISSUE TITLE
The context object is not shared between the activity and orchestrator dispatcher middlewares

### ISSUE THREAD

**Issue by govenkat-ms** [Azure/durabletask#454]
So I need to pass the .NET activity id (generated by the orchestrator middleware) to the activity middleware, and set that id as the parent to the activity created in the latter. It looks as though there is no global context used in the orchestration, so how does one pass a correlation id from orchestrator to activity to sub-orchestration etc.?

**Comment by cgillum** [comment]
There isn't any feature that would allow you to pass context between orchestrations and activities (that I am aware of). Data needs to be manually passed from orchestrations to activity functions.

Is your goal to do end-to-end correlation? If so, that is currently work-in-progress: https://github.com/Azure/durabletask/pull/422 (additional background here: https://medium.com/@tsuyoshiushio/durable-functions-distributed-tracing-71426fe2246f).

**Comment by govenkat-ms** [comment]
Yes the goal is end-to-end correlation. Is there an ETA on this PR to the core package?

**Comment by Robulane** [comment]
@cgillum, This is needed not only for end-to-end correlation but also to set some a common context for an operation.
We log data in a different source and we want to log some attributes that we set when we create an orchestration.

**Comment by jviau** [comment]
See my sample here for how to put this kind of data into an orchestration: https://github.com/jviau/durabletask-hosting/tree/master/samples/DurableTask.Samples/SessionData

It is a bit of a hack, but works great for the ServiceBus and Emulator implementation at least. Azure table storage had issues, but I checked in a fix for it recently so that should work as well.

**Comment by Robulane** [comment]
Unfortunately, this code doesn't work with DurableTask.AzureStorage.

MessageManager is using Json serialization:

            this.taskMessageSerializerSettings = new JsonSerializerSettings
            {
                TypeNameHandling = TypeNameHandling.Objects,
#if NETSTANDARD2_0
                SerializationBinder = new TypeNameSerializationBinder(),
#else
                Binder = new TypeNameSerializationBinder(),
#endif
            };

**Comment by jviau** [comment]
@Robulane I don't think the new AzureStorage package has been published yet. Here is the fix I checked in: https://github.com/Azure/durabletask/pull/492

**Comment by Robulane** [comment]
@jviau, shouldn't we have JsonExtensionData instead of IExtensibleDataObject? (this is the main issue for me as I can workaround type issue by writing directly into dictionary of Jtokens).

**Comment by jviau** [comment]
@Robulane you are right - json serialization won't respect `IExtensibleDataObject`. I had assumed it did because Newtonsoft supports `[DataContract]` and `[DataMember]` attributes. Will probably need to open an issue on AzureStorage provider to correctly honor `IExtensibleDataObject`.
